src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -163,38 +62,63 @@ function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: strin
   )
 }
 
-function useActiveWalletProvider(): Web3Provider | undefined {
-  return getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+interface ActiveWeb3ProviderProps {
+  provider?: Eip1193Provider | JsonRpcProvider
+  jsonRpcEndpoint?: string | JsonRpcProvider
+}
+
+export function ActiveWeb3Provider({
+  provider: propsProvider,
+  jsonRpcEndpoint,
+  children,
+}: PropsWithChildren<ActiveWeb3ProviderProps>) {
+  const metaMaskConnection = useMemo(
+    () => toWeb3Connection(initializeConnector<MetaMask>((actions) => new MetaMask({ actions }))),
+    []
+  )
+  const walletConnectConnectionQR = useMemo(() => getWalletConnectConnection(false, jsonRpcEndpoint), [jsonRpcEndpoint]) // WC via tile QR code scan
+  const walletConnectConnectionPopup = useMemo(
+    () => getWalletConnectConnection(true, jsonRpcEndpoint),
+    [jsonRpcEndpoint]
+  ) // WC via built-in popup
+
+  const networkConnection = useMemo(() => {
+    let network: string[] | JsonRpcProvider[]
+    if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
```

Comment:
why is this necessary? if the passed in value is a string or JsonRpcProvider should be all set. if it's undefined you should skip making a network connection or error

Reply:
if I just do `const urlMap = { [SupportedChainId.MAINNET]: [jsonRpcEndpoint] }`, we get type errors

<img width="667" alt="Screen Shot 2022-07-01 at 4 35 53 PM" src="https://user-images.githubusercontent.com/27475332/176964793-3367fe7e-22db-49ec-a31b-2783e3837754.png">

Reply:
good point on the undefined -- but if I don't handle these separately I get type errors:
<img width="632" alt="Screen Shot 2022-07-01 at 4 39 26 PM" src="https://user-images.githubusercontent.com/27475332/176965165-4902be9d-b27e-44eb-8a1a-0851cc350064.png">

Reply:
you need it to be typed as `(string | JsonRpcProvider)[]`

Reply:
right but Network's urlMap only takes in `JsonRpcProvider | url | url[] | JsonRpcProvider[]`

So essentially I need `networkRpc` to be `string[] | JsonRpcProvider[]`, not `(string | JsonRpcProvider)[]`

Reply:
okay i see